### PR TITLE
TECH-325 : clean code and refacto java

### PR DESCRIPTION
**Pull request by jsinovassin**
https://jira.jahia.org/browse/TECH-325

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -67,7 +67,7 @@
 
     <properties>
         <jahia-depends>jahia-authentication</jahia-depends>
-        <export-package>org.jahia.modules.jahiaoauth.service,org.jahia.modules.jahiaoauth.action,com.github.scribejava.*</export-package>
```
Comment (Taybou) [inline pom.xml:70]: question: why did you remove this package export?

Comment (tleclere-jahia) [reply pom.xml:70]: I don't know why this package was removed. This package is necessary to extend jahia-oauth in a custom module.

File: src/main/java/org/jahia/modules/jahiaoauth/action/ConnectToOAuthProvider.java [hunk 2]
```diff
@@ -2,23 +2,23 @@
...
+ * <p>
+ * This file is part of a Jahia's Enterprise Distribution.
+ * <p>
+ * Jahia's Enterprise Distributions must be used in accordance with the terms
+ * contained in the Jahia Solutions Group Terms & Conditions as well as
+ * the Jahia Sustainable Enterprise License (JSEL).
+ * <p>
+ * For questions regarding licensing, support, production usage...
+ * please contact our team at sales@jahia.com or go to http://www.jahia.com/license.
+ * <p>
```
Comment (tdraier) [inline src/main/java/org/jahia/modules/jahiaoauth/action/ConnectToOAuthProvider.java:21]: Please restore the correct license header (and ideally replace `/**` by `/*`)

File: pom.xml (hunk 1 above)
Comment (tdraier) [reply pom.xml:70]: yes, we cannot remove it

File: src/main/java/org/jahia/modules/jahiaoauth/connectors/FranceConnectConnectorImpl.java [hunk 3]
```diff
@@ -24,49 +24,22 @@
...
 import org.jahia.modules.jahiaoauth.service.OAuthConnectorService;
 
-import java.io.IOException;
-import java.util.List;
 import java.util.Map;
 
-public class FranceConnectConnectorImpl implements OAuthConnectorService {
+public class FranceConnectConnectorImpl extends Connector implements OAuthConnectorService {
 
     private Map<String, String> protectedResourceUrl;
```
Comment (tdraier) [inline src/main/java/org/jahia/modules/jahiaoauth/connectors/FranceConnectConnectorImpl.java:33]: This field is shading the `protectedResourceUrl` field from Connector class with different type and visibility. You should use another name.

File: src/main/java/org/jahia/modules/jahiaoauth/connectors/GoogleConnectorImpl.java [hunk 4]
```diff
@@ -2,71 +2,28 @@
  * ==========================================================================================
  * =                            JAHIA'S ENTERPRISE DISTRIBUTION                             =
  * ==========================================================================================
- *
- *                                  http://www.jahia.com
- *
+ * <p>
```
Comment (tdraier) [inline src/main/java/org/jahia/modules/jahiaoauth/connectors/GoogleConnectorImpl.java:5]: invalid license header

File: src/main/java/org/jahia/modules/jahiaoauth/service/JahiaOAuthService.java [hunk 5]
```diff
@@ -55,15 +55,6 @@
...
 public interface JahiaOAuthService {
 
-    /**
-     * This method will get the authorization URL so a connector can display the authentication popup to the user
-     *
-     * @param config    The oauth config for the connector
-     * @param sessionId String user session ID to be able to identify the token on the callback the session ID of the user is added to the request
-     * @return String authorization URL
-     */
-    String getAuthorizationUrl(ConnectorConfig config, String sessionId);
```
Comment (tdraier) [inline src/main/java/org/jahia/modules/jahiaoauth/service/JahiaOAuthService.java:65]: Are you sure you want to delete a public method that was released and potentially used ?

**Review by tdraier — APPROVED** [review]
